src/main/java/org/jenkinsci/plugins/docker/workflow/client/WindowsDockerClient.java

```diff
@@ -46,7 +47,11 @@ public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNu
         }
         for (Map.Entry<String, String> variable : containerEnv.entrySet()) {
             argb.add("-e");
-            argb.addMasked(variable.getKey()+"="+variable.getValue());
+            if (variable.getKey().contains("") || variable.getValue().contains(" ")) {
+                argb.addMasked(WindowsUtil.quoteArgument(variable.getKey() + "=" + variable.getValue()));
+            } else {
+                argb.addMasked(variable.getKey() + "=" + variable.getValue());
+            }
```

Comment:
That said, `WindowsUtil.quoteArgument` already [checks for a space or quote](https://github.com/jenkinsci/jenkins/blob/449c5aced523a6e66fe3d6a804e5dbfd5c5c67c6/core/src/main/java/hudson/os/WindowsUtil.java#L54) before doing anything, so I think this can be simplified:

```suggestion
            argb.addMasked(WindowsUtil.quoteArgument(variable.getKey() + "=" + variable.getValue()));
```

Reply:
Yes, I missed that. Thanks!